src/controllerclientimpl.cpp

```diff
@@ -1787,21 +1785,16 @@ void ControllerClientImpl::_UploadDataToController(const std::vector<unsigned ch
 
 void ControllerClientImpl::_DeleteFileOnController(const std::string& desturi)
 {
-    CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_CUSTOMREQUEST, NULL, "DELETE");
-    CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_HTTPHEADER, NULL, _httpheadersjson);
-    CURL_OPTION_SAVE_SETTER(_curl, CURLOPT_URL, NULL, desturi.c_str());
-    CURL_PERFORM(_curl);
-    long http_code = 0;
-    CURL_INFO_GETTER(_curl, CURLINFO_RESPONSE_CODE, &http_code);
-    // 204 is when it overwrites the file?
-    if( http_code != 204 ) {
-        if( http_code == 400 ) {
-            throw MUJIN_EXCEPTION_FORMAT("delete failed with HTTP status %s, perhaps file exists already?", desturi%http_code, MEC_HTTPServer);
-        }
-        else {
-            throw MUJIN_EXCEPTION_FORMAT("delete failed with HTTP status %s", desturi%http_code, MEC_HTTPServer);
-        }
+    MUJIN_LOG_DEBUG(str(boost::format("delete %s")%desturi))
+
+    // the dest filename of the upload is determined by stripping the leading _basewebdavuri
+    if( desturi.size() < _basewebdavuri.size() || desturi.substr(0,_basewebdavuri.size()) != _basewebdavuri ) {
+        throw MUJIN_EXCEPTION_FORMAT("trying to upload a file outside of the webdav endpoint is not allowed: %s", desturi, MEC_HTTPServer);
     }
+    std::string filename = desturi.substr(_basewebdavuri.size());
```

Comment:
Is `_basewebdavuri` here `/u/username`?

If so, it feels weird that we are converting from `mujin:/abc.mujin.dae` first to `/u/username/abc.mujin.dae` then to `abc.mujin.dae'

Reply:
> Is _basewebdavuri here /u/username?

Yes.

_PrepareDestinationURI_UTF8 needs to be rewritten. However the customer needs DeleteFileOnController_UTF8 API as soon as possible...

Reply:
...Actually, what should happen if I specify something not-including "mujin:/" as desturi?

Reply:
If the customer needs the API as soon as possible, I am okay with current changes as long as it works.